package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/sh
+# Copyright 2016-2017 Christof Schulze <christof@christofschulze.com>
+# Licensed to the public under the Apache License 2.0.
+
+. /lib/functions.sh
+. ../netifd-proto.sh
+init_proto "$@"
+
+proto_gluon_wireguard_init_config() {
+  no_device=1
+  available=1
+  renew_handler=1
+}
+
+proto_gluon_wireguard_renew() {
+  local config="$1"
+  echo wireguard RENEW: $@
+  ifdown $config
+  ifup $config
+}
+
+proto_gluon_wireguard_setup() {
+  local config="$1"
+  ifname=$(uci get network.$config.ifname) # we need uci here because nodevice=1 means the device is not part of the ubus structure
+
+  if [[ $(wg show all latest-handshakes |wc -l) -ge $(uci get wireguard.mesh_vpn_backbone.peer_limit) ]]; then
+    echo "not establishing another connection, we already have  $(uci get wireguard.mesh_vpn_backbone.peer_limit) connections." >&2
+    ip link del $ifname
+    ifdown $config
+    exit 1
+  fi
+
+  (
+  flock -n 9
+
+  if [ $(uci get wireguard.mesh_vpn_backbone.enabled) == 1 ]; then
+    ip link del $ifname
+    ip link add dev $ifname type wireguard
+    ip link set mtu $(uci get network.$config.mtu) dev $ifname
+    ip link set multicast on dev $ifname
+
+    group_peers=$(uci show wireguard|grep mesh_vpn_backbone_peer|grep =peer|cut -d"." -f2|cut -d= -f1)
+
+    for i in $group_peers
+    do
+      if [ $ifname == $(uci get wireguard.$i.ifname) ]; then
```

Comment:
Why don't you put the $ifname in quotation marks here?
And why do you only use single square brackets?

Reply:
actually setting multicast is required. The whole thing won't work without it.

You are right about the quotes, I added them.